Segfault on large account

When running aws-nuke against a large account with many resources, aws-nuke dies during the resource enumeration (specifically Lambda functions) with this error:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc4ca02]

goroutine 37 [running]:
github.com/rebuy-de/aws-nuke/resources.(*RDSNuke).ListInstances(0xc4202021e0, 0xc420054b40, 0xc4215bdfb0, 0xfacb01, 0xca7ed2, 0xc420189101)
	/go/src/github.com/rebuy-de/aws-nuke/resources/rds-instances.go:26 +0x152
github.com/rebuy-de/aws-nuke/resources.(*RDSNuke).ListInstances-fm(0xd759e0, 0xc420054b40, 0xc4215bdfb0, 0xe, 0x0)
	/go/src/github.com/rebuy-de/aws-nuke/resources/listers.go:95 +0x2a
github.com/rebuy-de/aws-nuke/cmd.Scan.func1(0xc420116180, 0xc4204003b0, 0xc420054b40)
	/go/src/github.com/rebuy-de/aws-nuke/cmd/scan.go:22 +0x77
created by github.com/rebuy-de/aws-nuke/cmd.Scan
	/go/src/github.com/rebuy-de/aws-nuke/cmd/scan.go:39 +0x9e

Happy to help.

- I think I see what some of the issue, based on the RDS query you provided.   RDS SQL Server instances do not contain a "DBName" key.  The result is nothing returned at the CLI, even though there are a dozen SQL Server instances.

- It does die each time, in the same spot.

- Running `aws-nuke with --target LambdaFunction` again dies at the same spot, but with a slightly different error:

```
./aws-nuke -c config.yaml --profile sandbox --target LambdaFunction
... snip ...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc4ca02]
goroutine 44 [running]:
github.com/rebuy-de/aws-nuke/resources.(*RDSNuke).ListInstances(0xc4202021e0, 0xc420184780, 0xc4202effb0, 0xfacb01, 0xca7ed2, 0xc42038bb01)
	/go/src/github.com/rebuy-de/aws-nuke/resources/rds-instances.go:26 +0x152
github.com/rebuy-de/aws-nuke/resources.(*RDSNuke).ListInstances-fm(0xd759e0, 0xc420184780, 0xc4202effb0, 0xe, 0x0)
	/go/src/github.com/rebuy-de/aws-nuke/resources/listers.go:95 +0x2a
github.com/rebuy-de/aws-nuke/cmd.Scan.func1(0xc42012a180, 0xc4203fc2d0, 0xc420184780)
	/go/src/github.com/rebuy-de/aws-nuke/cmd/scan.go:22 +0x77
created by github.com/rebuy-de/aws-nuke/cmd.Scan
	/go/src/github.com/rebuy-de/aws-nuke/cmd/scan.go:39 +0x9e
```

--- a/resources/rds-instances.go
+++ b/resources/rds-instances.go
@@ -6,9 +6,8 @@ import (
 )
 
 type RDSInstance struct {
-	svc  *rds.RDS
-	id   string
-	name string
+	svc *rds.RDS
+	id  string
 }
 
 func (n *RDSNuke) ListInstances() ([]Resource, error) {
@@ -21,9 +20,8 @@ func (n *RDSNuke) ListInstances() ([]Resource, error) {
 	resources := make([]Resource, 0)
 	for _, instance := range resp.DBInstances {
 		resources = append(resources, &RDSInstance{
-			svc:  n.Service,
-			id:   *instance.DBInstanceIdentifier,
-			name: *instance.DBName,
+			svc: n.Service,
+			id:  *instance.DBInstanceIdentifier,
 		})
 	}
 
@@ -45,5 +43,5 @@ func (i *RDSInstance) Remove() error {
 }
 
 func (i *RDSInstance) String() string {
-	return i.name
+	return i.id
 }

